pyro/infer/autoguide/guides.py

```diff
@@ -421,13 +421,9 @@ def _unpack_latent(self, latent):
         batch_shape = latent.shape[:-1]  # for plates outside of _setup_prototype, e.g. parallel particles
         pos = 0
         for name, site in self.prototype_trace.iter_stochastic_nodes():
-            constrained_shape = site["value"].shape
             unconstrained_shape = self._unconstrained_shapes[name]
             size = _product(unconstrained_shape)
-            event_dim = site["fn"].event_dim + len(unconstrained_shape) - len(constrained_shape)
```

Comment:
I may be missing context as to why we need this, so please let me know if I have oversimplified! This seems to pass existing tests and a new one that I added.

Reply:
The change looks fine to me but I might miss some context too.